pkg/newlog/cmd/newlogd.go

```diff
@@ -740,41 +779,33 @@ func remNonPrintable(str string) string {
 
 // writelogFile - a goroutine to format and write log entries into dev/app logfiles
 func writelogFile(logChan <-chan inputEntry, moveChan chan fileChanInfo) {
-
-	if _, err := os.Stat(collectDir); os.IsNotExist(err) {
-		if err := os.MkdirAll(collectDir, 0755); err != nil {
-			log.Fatal(err)
-		}
-	}
+	// get EVE version and partition, UUID may not be available yet
+	getEveInfo()
 
 	// move and gzip the existing logfiles first
-	logmetaData = findMovePrevLogFiles(moveChan)
+	findMovePrevLogFiles(moveChan)
 
-	// new logfile
-	devlogFile := startTmpfile(collectDir, devPrefix, false)
-	defer devlogFile.Close()
+	// new file to collect device logs for upload
+	devStatsUpload := initNewLogfile(collectDir, devPrefixUpload, "")
+	defer devStatsUpload.file.Close()
 
-	var fileinfo fileChanInfo
-	var devStats statsLogFile
+	// new file to collect device logs to keep on device
+	devStatsKeep := initNewLogfile(collectDir, devPrefixKeep, "")
+	defer devStatsKeep.file.Close()
+	devStatsKeep.notUpload = true
+
+	logmetrics.DevMetrics.LatestAvailableLog = time.Now()
```

Comment:
This one will change on every reboots of the device, but the older log entries can still stay on the disk, so it won't reflect the earliest one. Also this 'Latest' can be the most recent timestamp, maybe something like 'oldest_saved' or something like that

Reply:
that's true, very nice catch!!

Reply:
Something else to watch out for are devices (like RPi4) with no battery-backed up clock, which always boot on Jan 1st 1970 UTC. ntpd/chronos sets the clock, but if the network is out the agents including newlogd start before the time is set.

Reply:
@eriknordmark I'm not sure how to address that, because the logs will be then also dated Jan 1st 1970... we cannot wait until the time is set correctly and only then start logging